qiime2/plugins.py

```diff
@@ -37,6 +37,13 @@ def _init_metadata_(self, ref, factory):
         self._init_data_refs[ref] = factory
         return ref
 
+    def _init_data_collection_(self, ref, collection_type, *records):
+        self._init_data_refs[ref] = ref
+        collection = sorted([i.ref for i in records])
+        collection = collection_type(collection)
+        t = str(collection).replace("'", "")
+        return t
```

Comment:
Do you know, off the top of your head, how to consistently sort all possible inputs to `sorted` here?  I keep getting inconsistent test results, e.g. `{single_int1, single_int2}` and then a failing test case when collection ends up being `{single_int2, single_int1}`. [This](https://www.tutorialspoint.com/How-to-correctly-sort-a-string-with-a-number-inside-in-Python) didn't work, but I haven't dug any deeper.  Thought I'd check with you first.

Reply:
I think the issue is you're going from a sorted Iterable (list, L42), no an unsorted Iterable (set, L43). Set's don't have an order (https://docs.python.org/3.6/library/stdtypes.html#set-types-set-frozenset):

> A set object is an unordered collection of distinct hashable objects.

---

I think you might have to get a little more involved in the templating step here, maybe repr out a sorted list, then do a string substitution to replace square brackets with curlies?

Alternatively, you could template out something like this:

```python
x=set(['a', 'b', 'c', 'd'])
```

that uses the explicit set/list constructor.

Reply:
Oh, duhhh,  I don't even need the actual container type here 🤦‍♂️